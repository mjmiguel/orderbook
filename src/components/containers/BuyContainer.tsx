import React, { FC } from 'react';

type BuyProps = {};

const BuyContainer:FC = (props: BuyProps) => {
  return (
    <>
      <h1>Buy Container</h1>
      <div>
        <ul>
          <li>onething</li>
          <li>onething</li>
          <li>onething</li>
        </ul>
      </div>
    </>
  );

}

export default BuyContainer;