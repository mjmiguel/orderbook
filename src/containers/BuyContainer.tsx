import React, { FC } from 'react';
import fetch from 'node-fetch';

type BuyProps = {};

const BuyContainer:FC = (props: BuyProps) => {

  const getBooks = (exchange: string):void => {
    console.log('getbooks fired')
    fetch(`/api/${exchange}`)
      .then((res) => res.json())
      .then((data) => {
        console.log('got something back?', data);
      }).catch((e) => {
        console.log('there was an error', e);
      })
  }

  return (
    <>
      <h1>Buy Container</h1>
      <div>
        hello
      </div>
      <div>
        <button onClick={() => {getBooks('polo')}}>get Poloniex books</button>
      </div>
      <div>
        <button onClick={() => {getBooks('bitt')}}>get Bittrex books</button>
      </div>
    </>
  );

}

export default BuyContainer;