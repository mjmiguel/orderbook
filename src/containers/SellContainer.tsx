import React, { FC } from 'react';

type SellProps = {};

const SellContainer:FC = (props: SellProps) => {
  return (
    <>
      <h1>Sell Container</h1>
      <div>
        <ul>
          <li>onething</li>
          <li>onething</li>
          <li>onething</li>
        </ul>
      </div>
    </>
  );

}

export default SellContainer;