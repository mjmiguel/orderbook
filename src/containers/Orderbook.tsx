import React, { FC, useEffect, useState } from 'react';
import { useInterval } from '../hooks/hooks';
import BuyContainer from './BuyContainer';
import SellContainer from './SellContainer';

export const orderbookContext = React.createContext(null)

type OrderbookProps = {}

const Orderbook: FC<OrderbookProps> = (props: OrderbookProps) => {
  const [bids, setBids] = useState(null);
  const [asks, setAsks] = useState(null);

  // Fetch combined orderbooks for BTC_ETH markets
  const getBooks = ():void => {
    fetch(`/api`)
    .then((res) => res.json())
    .then((data) => {
      if (data.bid && data.ask) {
        setBids(data.bid);
        setAsks(data.ask);
      }
    }).catch((error) => {
      console.log('there was an error in /api fetch', error);
    })
  }

  // getBooks at an interval
  useInterval(getBooks, 5000);

  return (
    <>
      <h1>ORDERBOOK</h1>
      <orderbookContext.Provider value={{bids, asks}}>
        <BuyContainer />
        <SellContainer />
      </orderbookContext.Provider>
    </>
  );
}

export default Orderbook;