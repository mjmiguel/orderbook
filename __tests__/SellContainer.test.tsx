import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';

import SellContainer, { SellProps } from '../src/containers/SellContainer';
import { orderbookContext } from "../src/containers/Orderbook";

// assign default props and return rendered comp
function renderSellContainer(props: SellProps = {}) {
  const defaultProps: SellProps = { };
  return render(
    <orderbookContext.Provider value={[]}>
      <SellContainer {...defaultProps} {...props} />
    </orderbookContext.Provider>
  );
}

describe("<SellContainer />", () => {

  // get rendered component
  const { container, getByText } =  renderSellContainer();

  test(`should render an element containing "ASK`, async () => {
    expect(getByText('ASK')).toBeInTheDocument();
  });

  test(`should snapshot should match previous`, async () => {
    expect(container).toMatchSnapshot();
  });
});