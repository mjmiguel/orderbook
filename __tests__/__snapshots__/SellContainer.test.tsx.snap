// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SellContainer /> should snapshot should match previous 1`] = `<div />`;
